File: leetcode-submissions/1930_Unique_Length-3_Palindromic_Subsequences.ts
function countPalindromicSubsequence(s: string): number {
    const letters:Set<string>=new Set(s);
    let count:number =0;
    for(const letter of letters){
        const firstIndex=s.indexOf(letter);
        const lastIndex=s.lastIndexOf(letter);
        const chars:Set<string>=new Set();
        for(let i:number=firstIndex+1;i<lastIndex;i++)
        {
            chars.add(s[i]);
        }
        count+=chars.size;

    }
return count;
};

File: leetcode-submissions/1930_Unique_Length-3_Palindromic_Subsequences.cpp
class Solution {
public:
    int countPalindromicSubsequence(string s) {
        unordered_set<char>letters(s.begin(),s.end());
        int count =0;
        for(char letter:letters){
            //then we need to check the first and last index of the characters 
            int firstIndex=s.find(letter);
            int lastIndex=s.rfind(letter);
            unordered_set<char>chars;
            for(int i=firstIndex+1;i<lastIndex;i++){
                chars.insert(s[i]);
            }
            count+=chars.size();
        }
        return count ;

    }
};